src/engine/cuecontrol.cpp

```diff
@@ -711,7 +712,7 @@ void CueControl::cueCDJ(double v) {
     // If play is pressed while holding cue, the deck is now playing. (Handled in playFromCuePreview().)
 
     QMutexLocker lock(&m_mutex);
-    bool playing = m_pPlay->toBool();
+    bool playing = m_pPlay->toBool() && !getEngineBuffer()->getScratching();
```

Comment:
Did you consider the opposite case as well: A track is paused but scratched by the platter.  
In this case the cue point will be set on a somehow random position.

I think you actually are interested in the question if there is any track transport active or not. 
It looks like this can be checked by reading EngineBuffer::m_rate_old